File: src/main/java/at/searles/parsing/tokens/TokenParser.java
package at.searles.parsing.tokens;

import at.searles.lexer.Tokenizer;
import at.searles.parsing.*;
import at.searles.parsing.printing.ConcreteSyntaxTree;

public class TokenParser implements Parser<CharSequence> {

    private final int tokenId;
    private final Tokenizer tokenizer;
    private final boolean exclusive;

    public TokenParser(int tokenId, Tokenizer tokenizer, boolean exclusive) {
        this.tokenId = tokenId;
        this.tokenizer = tokenizer;
        this.exclusive = exclusive;
    }

    @Override
    public boolean recognize(ParserStream stream) {
        return stream.parseToken(tokenizer, tokenId, exclusive) != null;
    }

    @Override
    public CharSequence parse(ParserStream stream) {
        return stream.parseToken(tokenizer, tokenId, exclusive);
    }

    @Override
    public ConcreteSyntaxTree print(CharSequence seq) {
        return ConcreteSyntaxTree.fromCharSequence(seq);
    }

    public String toString() {
        return String.format("<%d>", tokenId);
    }
}

File: src/main/java/at/searles/parsing/tokens/TokenParser.kt
package at.searles.parsing.tokens

import at.searles.lexer.Tokenizer
import at.searles.parsing.Parser
import at.searles.parsing.ParserStream
import at.searles.parsing.printing.ConcreteSyntaxTree

class TokenParser(val tokenId: Int, val tokenizer: Tokenizer) : Parser<CharSequence> {
    override fun recognize(stream: ParserStream): Boolean {
        return stream.parseToken(tokenizer, tokenId) != null
    }

    override fun parse(stream: ParserStream): CharSequence? {
        return stream.parseToken(tokenizer, tokenId)
    }

    override fun print(item: CharSequence): ConcreteSyntaxTree? {
        return ConcreteSyntaxTree.fromCharSequence(item)
    }

    override fun toString(): String {
        return "<$tokenId>"
    }

}